### PR TITLE
Start on creating component library, reorganize some things

### PR DESCRIPTION
**Pull request by Dacello**
Description:
- Adds package.json with react and react-native dependencies, preparing us to be able to publish this as an npm package as well as actually create react components to be used
- Creates a new components directory with a Button and Typography component to start
- Refactors the way we organize our base styles
  - We have a `components` directory in which each component gets its own directory with a `styles.js` and `index.js`, where `index.js` contains the component itself and `styles.js` contains the actual RN `StyleSheet` used in the component.
  - We then have a `variables` directory in which we have files for colors, spacing, fonts, typography. These are strictly reusable variables, but dont have any actual style properties included
  - I think its best that we separate out variables from actual component styles as much as possible.

NOTE: Would be nice to add Storybook to this so it can have a baked in style guide of the components we add. This will require adding an `index.stories.js` to each of the component directories. This is fairly straightforward and has been done on other projects

### CODE REVIEW

File: variables/index.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// these are variables that don't rely on other styles
+import * as Colors from './colors'
```
Comment (brianberlin) [inline variables/index.js:2]: i think these should be lower case since they are not classes or components.

File: components/Button/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import styles from './styles'
+
+export default function Button ({
+  onPress,
+  title,
+  isLoading,
+  variant = 'primary',
+  containerStyle = {},
+  buttonStyle = {},
+  textStyle = {}
```
Comment (brianberlin) [inline components/Button/index.js:14]: I'm still trying to wrap my head around how a theme context/provider might help with things like this where we want to override some defaults. I think what we want to do is wrap all the variables and styles for each component into a theme context that can be changed and overridden.

A button is a button normally but sometimes you gotta put one in a weird spot so if we put theming variables and the styles they create into a react context then i think we could deal with the edge cases like...

```
<AppRoot>
  <Provider theme={defaultTheme}>
    <Button />
    <Provider theme={weirdButtonTheme}>
      <Button />
    </Provider>
</AppRoot>
```

I'm pulling a lot of this from [material-ui](https://material-ui.com/customization/theming/) which also includes a good list of [default theme variables](https://material-ui.com/customization/default-theme/)

**Review by brianberlin — APPROVED** [review]

File: variables/index.js (hunk 1 above)
Comment (Dacello) [reply variables/index.js:2]: Yea i agree -- was trying to maintain some consistency with Laura's initial groundwork but will update.

**Comment by jwebber18** [comment]
- Changed dependencies to peer dependencies to make this installable as a package
- Changed the root export to only export components (can still import variables by importing from harmonium-rn/variables)
- Tested the button and typography components out in a test app, fixed a few import problems / misc bugs

**Pull request merged**